### PR TITLE
[Mod]: : Missing Input Validation - Edge Case Handling

### LINKED ISSUE #212: Missing Input Validation - Edge Case Handling
Issue Description:
Several functions lack comprehensive input validation, particularly for edge cases like zero amounts, invalid timestamps, and boundary conditions. This could lead to unexpected behavior or potential exploits.

Current Problem:

// Missing validation for edge cases
fn vote(ref self: ContractState, pool_id: u256, option: felt252, amount: u256) {
    // No validation that amount > 0
    // No validation that pool_id exists before other checks
    // No validation for option format
}
PR Requirements:

Add comprehensive input validation for all public functions
Implement boundary condition checks
Add validation for zero amounts and invalid inputs
Create validation test suite
Update error messages for better user experience

### PR DESCRIPTION
**Pull request by ryzen-xp**
### PR Title  
**Fix: Add Input Validation & Edge Case Handling (#212)**  

### Description  
Adds comprehensive input validation across public functions:  
- Check `amount > 0`  
- Validate `pool_id` existence  
- Enforce valid `option` ranges  
- Handle invalid timestamps & boundary conditions  
- Improve error messages for clarity  

Includes a new **validation test suite** to cover zero/invalid inputs and edge cases.  

---  

Screenshot from 2025-08-28 21-36-04

### CODE REVIEW

File: src/base/errors.cairo [hunk 1]
```diff
@@ -114,4 +114,13 @@ pub mod Errors {
 
     /// @notice Error: The caller is not authorized to perform this action.
     pub const UNAUTHORIZED_CALLER: felt252 = 'Unauthorized Caller';
+
+    /// @notice Error: The pool_id is zero , it must greater then zero.
```
Comment (Akshola00) [inline src/base/errors.cairo:118]: gm great having predefined error messages here, makes it easier to handle changes

File: src/predifi.cairo [hunk 2]
```diff
@@ -430,7 +430,13 @@ pub mod Predifi {
         fn vote(ref self: ContractState, pool_id: u256, option: felt252, amount: u256) {
             self.pausable.assert_not_paused();
 
-            let pool = self.pools.read(pool_id);
+            // Input Validation
```
Comment (Akshola00) [inline src/predifi.cairo:433]: great work here also, more checks = more secure contract, 

@ryzen-xp pls create an impl security trait function and move this security impl into it, call the functions here, makes it more modular

File: tests/integration_test.cairo [hunk 3]
```diff
@@ -862,3 +862,132 @@ fn test_multiple_users_voting_and_natural_settlement() {
     dispute_contract.claim_reward(pool_id);
     stop_cheat_caller_address(dispute_contract.contract_address);
 }
+
+
+#[test]
```
Comment (Akshola00) [inline tests/integration_test.cairo:867]: solid tests. great work

**Review by Akshola00 — CHANGES_REQUESTED** [review]
overall looking great, although you only did more validations for the function mentioned as example on the issue description 

`Several functions lack comprehensive input validation` if you could check for other functions what need more validation and do validation implementations here that would be great

if there isnt then with the requested changes it should be good to go

**Review by Akshola00 — APPROVED** [review]
this is better now
nicely done !
